aries_cloudagent/core/event_bus.py

```diff
@@ -45,7 +48,7 @@ def __init__(self):
         """Initialize Event Bus."""
         self.topic_patterns_to_subscribers: Dict[Pattern, Sequence[Callable]] = {}
 
-    async def notify(self, profile: Profile, event: Event):
+    async def notify(self, profile: "Profile", event: Event):
```

Comment:
What do you think of having profile be an optional property of Event? Is there a specific reason that all events need a profile?

Reply:
Profile represents the wallet that emitted and should therefore process the event. This really only comes into play for multi-tenanted agents where one wallet manages many wallets. Omitting profile, the event handlers wouldn't have a handle to the correct storage and wallet to process the event.

Reply:
Ok, makes sense that most events (in fact all of the ones we currently have) need a profile. However, do you think in the future we might need system wide events that are not specific to a wallet?

i.e.

`event_bus.notify(context.profile, Event("topic", payload))`
becomes
`event_bus.notify(Event("topic", payload, profile=context.profile))`

The annoying thing will be that most subscribers will need to assert that profile is present when handling the event.

Reply:
Fortunately, I think this is already kind of being taken care of given that there is the concept of a "root" profile on startup at least. Beyond that, I can't think of many scenarios where we would need system wide event since basically everything in ACA-Py is scoped to a wallet and storage, which must be provided by the profile. So I'm inclined to say that we skip it for now and, if it becomes obvious that is is needed in the future, we can worry about it then.

Reply:
Ok, sounds good